src/app.js

```
@@ -124,18 +112,18 @@ class SidebarItem extends React.Component {
 
 class TheModal extends React.Component {
   render() {
-    const { item } = this.props;
-    if (item.modal) {
-      const Modal = React.createElement(item.modal);
+    const { eso } = this.props;
+    const ContainedView = eso.getView && eso.getView();
```

Discussion, non-blocking: In the future, I imagine we'll want to fail early and loudly when something that doesn't have a `getView` is passed to `TheModal`. If not, then I suppose we'll want to extend what happens on line 124.

For now, this is a nice way to build ESOs incrementally without making the app unusable.

---

Yeah, I had the same thought.  Either have faith that it's okay, or throw a tantrum, not this halfway thing.  But I couldn't make up my mind.